README.md

```diff
@@ -1,8 +1,93 @@
-# haystack-experimental
-
 [![PyPI - Version](https://img.shields.io/pypi/v/haystack-experimental.svg)](https://pypi.org/project/haystack-experimental)
 [![PyPI - Python Version](https://img.shields.io/pypi/pyversions/haystack-experimental.svg)](https://pypi.org/project/haystack-experimental)
 [![Tests](https://github.com/deepset-ai/haystack-experimental/actions/workflows/tests.yml/badge.svg)](https://github.com/deepset-ai/haystack-experimental/actions/workflows/tests.yml)
 [![Project release on PyPi](https://github.com/deepset-ai/haystack-experimental/actions/workflows/pypi_release.yml/badge.svg)](https://github.com/deepset-ai/haystack-experimental/actions/workflows/pypi_release.yml)
 [![Hatch project](https://img.shields.io/badge/%F0%9F%A5%9A-Hatch-4051b5.svg)](https://github.com/pypa/hatch)
-[![Checked with mypy](https://www.mypy-lang.org/static/mypy_badge.svg)](https://mypy-lang.org/)
\ No newline at end of file
+[![Checked with mypy](https://www.mypy-lang.org/static/mypy_badge.svg)](https://mypy-lang.org/)
+
+# Haystack experimental package
+
+The `haystack-experimental` package provides Haystack users with access to experimental features without immediately
+committing to their official release. The main goal is to gather user feedback and iterate on new features quickly.
+For simplicity, every release of `haystack-experimental` will ship all the available experiments at that time.
```

Comment:
Shall we also leave a note here about the versioning scheme (CalVer)? Also, I think it would be good to add a sentence that this package is meant to be used with the latest release of the core package (or at least that we don't provide any guarantees about their compatibility with core versions that are older than the current version of the experimental package).

Reply:
Absolutely, let me investigate how painful will be CalVer compared to other solutions with the same goal and let's make it official.

Reply:
I think we should ditch CalVer, it adds a lot of complexity and I start doubting the benefits of it.